### DISCUSSION TITLE
.next/data always not found

### DISCUSSION THREAD

**Discussion by alfianpurnomo-elev** [vercel/next.js#24363] [Help]
Hi,
I have a problem deploying the application with pm2
but it goes well with vercel

i always get error
/_next/data/[build id]/file_name.js not found on console log

anyone can help please
im stuck with this problem

**Comment by akellbl4** [comment]
Hey
Could you please describe what you do for deployment?

**Reply by alfianpurnomo-elev** [reply]
in development i used pm2
first i clone repository
then running yarn install
then yarn build
after that i running 

pm2 start yarn --name "app_name" --interpreter bash -- start

after that i configure nginx to access .next folder

**Reply by akellbl4** [reply]
I think it's no need to configure nginx because when you run `next start` Next.js is serving all files for you

**Reply by alfianpurnomo-elev** [reply]
but for production cannto run on port 3000

**Reply by akellbl4** [reply]
Understood, you need to proxy not to files but to upstream.
The config should look like this:
```
server {
  ...

  location / {
    proxy_pass http://localhost:3000;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection 'upgrade';
    proxy_set_header Host $host;
    proxy_cache_bypass $http_upgrade;
  }
}
